content/modeling/modeling_3.md

```diff
@@ -0,0 +1,117 @@
+- `consider_intrasession_channels`: Boolean. If false, only considers the channel at the start of the session (i.e. first page view). If true, considers multiple channels in the conversion session as well as historically.
+
+The default source schemas and tables used by the snowplow_fractribution package are:
+- *atomic_derived.snowplow_web_page_views* for the page_views data (page_views_source)
+- *atomic.events* for the Snowplow event data (conversions_source)
+
+If either of these differ in your warehouse, set the correct names as variables in your own dbt_project.yml, e.g.:
+-  `page_views_source`: `custom_schema_derived.snowplow_web_page_views`
+-  `conversions_source`: `custom_schema_derived.snowplow_ecommerce_transaction_interactions`
+
+For example, if you do not wish the package to query your `atomic.events` table to get the conversion value data, the `conversions_source` variable can point to a different table, such as the derived table `snowplow_ecommerce_transaction_interactions` created by the ecommerce model. **(TODO: Add link)**
+Another option would be to create your own incremental data model for transaction/conversion events, and add a reference to that model as the conversions_source variable, e.g. `conversions_source: {{ ref('custom_conversions_model') }}`
+
+You only need to set the variables for those that differ from the default.
+
+Below is an example snippet of these variables in your `dbt_project.yml`:
+
+```yml
+vars:
+  conversion_window_start_date: '2022-06-03'
+  conversion_window_end_date: '2022-08-01'
+  conversion_hosts: ['poplindata.com']
+  path_lookback_steps: 0
+  path_lookback_days: 30
+  path_transforms: [['Exposure', null]]
+  consider_intrasession_channels: false
+  conversions_source: 'atomic.sample_events_fractribution'
+```
+
+**Path Transform Options**
+
+Paths to conversion are often similar, but not identical. As such, path transforms reduce unnecessary complexity in similar paths before running the attribution algorithm.
+
+There are five path transform options available:
+- Exposure (default in this package): the same events in succession are reduced to one: A -> A -> B becomes A -> B. A compromise between first and unique.
+- Unique: all events in a path are treated as unique (no reduction of complexity). Best for smaller datasets (small lookback window) without a lot of retargeting.
+- First: keep only the first occurrence of any event: A -> B -> A becomes A -> B. Best for brand awareness marketing.
+- Frequency: keep a count of the events' frequency: A -> A -> B becomes A(2) -> B. Best when there is a lot of retargeting.
+- Recency: time bucket events depending on when they occurred before the conversion and collapse those in the same bucket (square braces indicate minutes before conversion event): A[15-30] -> B[1-5] -> A[1-5] -> B[1-5] becomes A[15-30] -> B[1-5] -> A[1-5]. Best for larger lookback windows.
```

Comment:
I have noticed `Recency` is not among UDFs, is it missing from the model or it should be removed from here? Also not sure if the example is correct here as there are 2 `A` groups and if I understand correctly it should be unique after collapsing (?).

Reply:
Oh good catch, thank you - I shouldn't have added this. Deleted. The example is a little confusing anyway - the Bs are collapsed because they are in the same bucket [1-5] but the As aren't because they are different time buckets [15-30] vs [1-5]